src/config.rs

```diff
@@ -429,6 +429,80 @@ pub struct DbOptions {
 }
 
 impl DbOptions {
+    /// Logs the current configuration and enabled features.
+    pub fn log_config(&self) {
```

Comment:
I think it might make sense for us to leverage the serialization module that was built for #263 instead of having this be manual. Otherwise we have to remember every time a new configuration is added to log it.

Reply:
@agavra I was thinking of just using `Debug`. Initially, I thought maybe `serde_json`, but `Debug` seemed more idiomatic? WDYT?

Reply:
Oh that makes sense to me too! I'm not super familiar with Rust, so long as it's enforced I'm happy. Only benefit from logging it in a serialized format is it makes it easier to repro an environment by just looking at the logs.

Reply:
Yea.. plus we can do `from_file` then... So yea, maybe JSON is best, so we can just copy/paste.